File: replace_colors.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    file = path.join(dir, file);
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) {
      results = results.concat(walk(file));
    } else if (file.endsWith('.jsx')) {
      results.push(file);
    }
  });
  return results;
}

const files = walk('frontend/src');

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  content = content.replace(/text-slate-[123]00/g, 'text-main');
  content = content.replace(/text-slate-[45]00/g, 'text-muted');
  fs.writeFileSync(file, content, 'utf8');
});

console.log('Done replacing text colors.');

File: replace_colors.py
import os
import glob
import re

files = glob.glob('frontend/src/**/*.jsx', recursive=True)

for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Replace bright text
    content = re.sub(r'text-slate-[123]00', 'text-main', content)
    # Replace muted text
    content = re.sub(r'text-slate-[45]00', 'text-muted', content)
    
    with open(file, 'w', encoding='utf-8') as f:
        f.write(content)

print("Done replacing text colors.")
